src/mhd_system.hpp

```diff
@@ -26,6 +30,294 @@ template <typename problem_t> class MHDSystem
 	enum varIndex_perDim {
 		bfield_index = Physics_Indices<problem_t>::mhdFirstIndex,
 	};
+
+	static void ComputeEMF(std::array<std::array<amrex::MultiFab, 2>, AMREX_SPACEDIM> &ec_mf_emf_comps, amrex::MultiFab const &cc_mf_cVars,
+			       std::array<amrex::MultiFab, AMREX_SPACEDIM> const &fcx_mf_cVars, std::array<amrex::MultiFab, AMREX_SPACEDIM> const &fcx_mf_fspds,
+			       int nghost_fc);
+
+	static void ReconstructTo(FluxDir dir, arrayconst_t &cState, array_t &lState, array_t &rState, const amrex::Box &reconstructRange,
+				  int reconstructionOrder);
 };
 
+template <typename problem_t>
+void MHDSystem<problem_t>::ComputeEMF(std::array<std::array<amrex::MultiFab, 2>, AMREX_SPACEDIM> &ec_mf_emf_comps, amrex::MultiFab const &cc_mf_cVars,
+				      std::array<amrex::MultiFab, AMREX_SPACEDIM> const &fcx_mf_cVars,
+				      std::array<amrex::MultiFab, AMREX_SPACEDIM> const &fcx_mf_fspds, const int nghost_fc)
+{
```

Comment:
There are a lot of very small (one-line or similar) GPU kernels in this function. This will have a significant negative performance impact. Once test problems are working, I would strongly recommend seeing if it's possible to combine this kernels in this function into a much smaller number of `ParallelFor`s.